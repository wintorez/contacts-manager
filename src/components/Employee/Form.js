import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Form, Button, Grid } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

const defaultState = {
  employeeId: null,
  firstName: '',
  lastName: '',
  email: '',
  telephone: '',
  department: ''
};

export default class extends Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired
  };

  static defaultProps = {
    employee: {}
  };

  state = {
    ...defaultState,
    ...this.props.employee
  };

  handleChange = (e, { name, value }) => this.setState({ [name]: value });

  handleSubmit = () => this.props.onSubmit(this.state);

  render() {
    const { loading } = this.props;
    const {
      employeeId,
      firstName,
      lastName,
      email,
      telephone,
      department
    } = this.state;

    return (
      <Form onSubmit={this.handleSubmit} disabled={loading}>
        <Grid columns="2" stackable>
          <Grid.Row>
            <Grid.Column>
              <Form.Input
                label="First Name"
                placeholder="First Name"
                defaultValue={firstName}
                name="firstName"
                onChange={this.handleChange}
                required
              />
            </Grid.Column>
            <Grid.Column>
              <Form.Input
                label="Last Name"
                placeholder="Last Name"
                defaultValue={lastName}
                name="lastName"
                onChange={this.handleChange}
                required
              />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column>
              <Form.Input
                label="Email"
                placeholder="Email"
                type="email"
                defaultValue={email}
                name="email"
                onChange={this.handleChange}
                required
              />
            </Grid.Column>
            <Grid.Column>
              <Form.Input
                label="Telephone"
                placeholder="Telephone"
                type="tel"
                defaultValue={telephone}
                name="telephone"
                onChange={this.handleChange}
              />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column>
              <Form.Input
                label="Department"
                placeholder="Department"
                defaultValue={department}
                name="department"
                onChange={this.handleChange}
              />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column>
              <Button type="submit" loading={loading}>
                Save
              </Button>{' '}
              or{' '}
              <Link to={employeeId ? `/employees/${employeeId}` : '/employees'}>
                Cancel
              </Link>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Form>
    );
  }
}
