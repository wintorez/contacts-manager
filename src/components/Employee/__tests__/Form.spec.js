import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Form from '../Form';

const defaults = {
  history: {},
  onSubmit: jest.fn(),
  employee: {
    employeeId: '1',
    firstName: 'Joe',
    lastName: 'Bloggs',
    email: 'jblogs@email.com',
    telephone: '+1-555-444-3333',
    department: 'Engineering'
  }
};

describe('<Edit />', () => {
  it('renders correctly with employee', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Form {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without employee', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Form {...defaults} employee={undefined} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Form {...defaults} loading />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
