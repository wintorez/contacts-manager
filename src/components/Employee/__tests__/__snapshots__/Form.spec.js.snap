// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit /> renders correctly when loading 1`] = `
<form
  action={undefined}
  className="ui form"
  disabled={true}
  onSubmit={[Function]}
>
  <div
    className="ui stackable two column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            First Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Joe"
              disabled={undefined}
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Last Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Bloggs"
              disabled={undefined}
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Email
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="jblogs@email.com"
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              tabIndex={undefined}
              type="email"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Telephone
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="+1-555-444-3333"
              disabled={undefined}
              name="telephone"
              onChange={[Function]}
              placeholder="Telephone"
              required={undefined}
              tabIndex={undefined}
              type="tel"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Department
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Engineering"
              disabled={undefined}
              name="department"
              onChange={[Function]}
              placeholder="Department"
              required={undefined}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <button
          className="ui loading button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
         
        or
         
        <a
          href="/employees/1"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</form>
`;

exports[`<Edit /> renders correctly with employee 1`] = `
<form
  action={undefined}
  className="ui form"
  disabled={undefined}
  onSubmit={[Function]}
>
  <div
    className="ui stackable two column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            First Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Joe"
              disabled={undefined}
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Last Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Bloggs"
              disabled={undefined}
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Email
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="jblogs@email.com"
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              tabIndex={undefined}
              type="email"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Telephone
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="+1-555-444-3333"
              disabled={undefined}
              name="telephone"
              onChange={[Function]}
              placeholder="Telephone"
              required={undefined}
              tabIndex={undefined}
              type="tel"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Department
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue="Engineering"
              disabled={undefined}
              name="department"
              onChange={[Function]}
              placeholder="Department"
              required={undefined}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <button
          className="ui button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
         
        or
         
        <a
          href="/employees/1"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</form>
`;

exports[`<Edit /> renders correctly without employee 1`] = `
<form
  action={undefined}
  className="ui form"
  disabled={undefined}
  onSubmit={[Function]}
>
  <div
    className="ui stackable two column grid"
  >
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            First Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue=""
              disabled={undefined}
              name="firstName"
              onChange={[Function]}
              placeholder="First Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Last Name
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue=""
              disabled={undefined}
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
              required={true}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="required field"
        >
          <label
            htmlFor={undefined}
          >
            Email
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue=""
              disabled={undefined}
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              tabIndex={undefined}
              type="email"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Telephone
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue=""
              disabled={undefined}
              name="telephone"
              onChange={[Function]}
              placeholder="Telephone"
              required={undefined}
              tabIndex={undefined}
              type="tel"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Department
          </label>
          <div
            className="ui input"
            content={undefined}
          >
            <input
              defaultValue=""
              disabled={undefined}
              name="department"
              onChange={[Function]}
              placeholder="Department"
              required={undefined}
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <button
          className="ui button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="submit"
        >
          Save
        </button>
         
        or
         
        <a
          href="/employees"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</form>
`;
